src/lambdas/sdk_analysis/src/lib/main.py

```diff
@@ -0,0 +1,131 @@
+from io import StringIO
+import logging
+
+import paramiko
+import pymysql
+from sshtunnel import SSHTunnelForwarder
+
+from .aws_utils import get_remote_aws_endpoints
+from .aws_utils import get_s3_object_contents
+from .aws_utils import get_ssm_secrets
+from .helpers import load_data_to_dataframe
+
+INFO_DICT = {}
+
+# set logger
+logging.basicConfig(format="%(asctime)s [%(levelname)s] %(name)s: %(message)s", level=logging.INFO)
+logger = logging.getLogger()
+
+# Queries
+insert_into_uploaded_s3_table = """
+    INSERT INTO uploaded_s3_objects(bucket, object_key, upload_started_at, uploading_computer_name)
+    VALUES (%s, %s, NOW(), %s);
+    """
+
+insert_into_mantarray_recording_sessions = """
+    INSERT INTO mantarray_recording_sessions(mantarray_recording_session_id, instrument_serial_number, backend_log_id, acquisition_started_at, length_centimilliseconds,
+    recording_started_at)
+    VALUES (%s, %s, %s, %s, %s, %s);
+    """
+
+insert_into_mantarray_raw_files = """
+    INSERT INTO mantarray_raw_files(bucket_id, well_index, length_centimilliseconds, recording_started_at, mantarray_recording_session_id)
+    VALUES (SELECT id FROM uploaded_s3_objects ORDER BY id DESC LIMIT 1, %s, %s, %s, %s);
+    """
+
+insert_into_s3_objects = """
+    INSERT INTO s3_objects(bucket_id, kilobytes, stored_at) VALUES (SELECT id FROM uploaded_s3_objects ORDER BY id DESC LIMIT 1, %s, %s);
+    """
+
+
+def handle_db_metadata_insertions(bucket: str, key: str, file, r):
+    """ Open an SSH tunnel and connect using a username and password. Query database.
+        Args:
+            file: .xlsx file containing aggregated metadata for recording
+            r: PlateRecording instance for individual well data
+    """
+
+    if not INFO_DICT:
+        set_info_dict()
+
+    with SSHTunnelForwarder(
```

Comment:
I'm not sure I understand what this is for, the ecs task that is inserting the metadata should be running in the same VPC as the database with the same security group so it should be able to access the db without an ssh tunnel.

Was it not able to do that in your testing?

Reply:
That makes sense.... I think I was just trying to run the files locally.